src/bytes.rs

```diff
@@ -275,6 +275,14 @@ impl CharIterator {
         result
     }
 
+    /// Read directly from bytes
+    pub fn read_from_bytes(&mut self, bytes: &[u8], e: Option<Encoding>) -> io::Result<()> {
```

Comment:
Reqwest (and pretty much every system) will return an array of bytes, so we should be able to read that. I'm not 100% sure about ownership and who needs to take it.. We don't want this to be copied a few times.